Wikipedia/Code/ArticleViewController.swift

```diff
@@ -152,7 +152,9 @@ class ArticleViewController: ViewController, HintPresenting {
 
         super.init(theme: theme)
         
-        self.schemeHandler.imageDidSuccessfullyLoad = imageDidSuccessfullyLoad
```

Comment:
There's a retain cycle here is because:

`self.schemeHandler.imageDidSuccessfullyLoad = imageDidSuccessfullyLoad`

is basically equivalent to:

`self.schemeHandler.imageDidSuccessfullyLoad = { self.imageDidSuccessfullyLoad() }`

In other words, strong `self` is being implicitly captured, which creates the retain cycle. It's a real shame though because I generally prefer the shorter syntax too (when it doesn't create a retain cycle 😀).